engine/evarutil.mli

```diff
@@ -126,6 +126,10 @@ val gather_dependent_evars : evar_map -> Evar.t list -> (Evar.Set.t option) Evar
     solved. *)
 val advance : evar_map -> Evar.t -> Evar.t option
 
+(** [reachable_from_evars sigma seeds evs] computes if one of [evs] is a descendent
+    of an evar of [seeds] by restriction or evar-evar unifications in [sigma]. *)
+val reachable_from_evars : evar_map -> Evar.Set.t -> Evar.Set.t -> bool
```

Comment:
Maybe it is just a dejavu but I seem to recall we discussed about this API and my impression was that what one wants is
```
val reachable_from_evars : evar_map -> Evar.Set.t -> Evar.Set.t 
```
And the compose the result with `Evar.Set.mem` in order to get the boolean.
Would it be possible to get this API? Or it would be inefficient.

Reply:
I don't remember that solution being discussed but it does make sense. The current implementation can stop early using an exception but basically builds a visited set that could just be completed and returned instead. I can try that as an alternative solution.

Reply:
My use case is to shelve goals (evars) that are reachable from already existing goals. Hence having the set, rather than a membership test, is probably simpler.

Reply:
Yes, that makes sense. Note that reachable_from_evars just computes a transitive closure by aliasing so it containts the defined evars as well, which is necessary for `check_evars`. It is easy to filter them out if necessary.

Reply:
I just pushed a new commit implementing that